lua/nvim-treesitter/configs.lua

```diff
@@ -421,6 +421,9 @@ function M.setup(user_data)
   end
 
   local ensure_installed = user_data.ensure_installed or {}
+  if type(config.ensure_installed) ~= "table" then
+    ensure_installed = config.ensure_installed
+  end
```

Comment:
Does this ever run? When `config.ensure_installed` is `{"lua", "vim", "vimdoc", ...}`

Reply:
This is intentionally only allowed to run when the value is a string, when `config.ensure_installed` is a table, continue to execute, and then it may merge with the new config, aka `user_data.ensure_installed`

Run this only when `config.ensure_installed` is a string, that is "all".

Reply:
but `config.ensure_installed` is always `{ "vim", "lua", "luadoc", ... }`

Reply:
No, the definition above is just the initial value, it's current `config.ensure_installed` value which is modified by users in https://github.com/nvim-treesitter/nvim-treesitter/blob/97ba59c6f532072e456b3d1f2413560e689c132b/lua/nvim-treesitter/configs.lua#L436

a simple proof by nvim-dap:

![image](https://github.com/nvim-treesitter/nvim-treesitter/assets/61115159/8455b6bb-fee1-4dfa-9286-f89d09b58735)